core_java/twitter/src/main/java/ca/jrvs/apps/twitter/DAO/TwitterService.java

```diff
@@ -0,0 +1,129 @@
+package ca.jrvs.apps.twitter.DAO;
+
+import ca.jrvs.apps.twitter.model.Coordinates;
+import ca.jrvs.apps.twitter.model.Tweet;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class TwitterService implements Service {
+
+  private CrdDao dao;
+
+  //@Autowired
+  public TwitterService(CrdDao dao) {
+    this.dao = dao;
+  }
+
+  @Override
+  public Tweet postTweet(Tweet tweet) {
+    //get Text
+    //get long/lat
+    if (tweet == null) {
+      throw new IllegalArgumentException("Value of Tweet is null");
+    }
+    Coordinates coordinates = tweet.getCoordinates();
+    int textLength = tweet.getText().length();
+    if (textLength > 140) {
+      throw new IllegalArgumentException("Text over 140 characters");
```

Comment:
Max tweet length is now 280, but changing that is not critical.
It may be worth your time to read [Twitter's page on counting characters](https://developer.twitter.com/en/docs/basics/counting-characters).